### ISSUE TITLE
Explore `ngrok` alternatives

### ISSUE THREAD

**Issue by HamedMP** [HamedMP/CursorLens#11]
Here's a good list of the alternatives. Cloudflare tunneling looks very interesting. 

https://github.com/anderspitman/awesome-tunneling

**Comment by shankscoder** [comment]
If you want to run this locally, do you even need ngrok? Is ngrok a baked in dependency or only to allow the remote access to the service?

**Comment by shankscoder** [comment]
Yup, thanks. I read the FAQ a bit more closely and noticed this😅 

Ok cool, I'll try to play around with a couple of different Docker Compose stacks to address this. I really don't like to depend on a third-party service for what is essentially local resolution.

FWIW, on a Mac, if you use Orbstack as your Docker daemon, the HTTPS is baked in by default to allow containers to be accessible with SSL. More here: https://docs.orbstack.dev/features/https

**Comment by tylersmalley** [comment]
You can use Tailscale serve (`tailscale serve 3000`), it's free, provides an HTTPS interface that isn't available to the public internet which sounds like it achieves what is required. If you need it to be available on the public internet you can use Funnel (`tailscale funnel 3000`)

**Comment by HamedMP** [comment]
That's great @tylersmalley, I used tailscale for creating private network in my dev team. I'll explore adding it to our Docker file
